docs/Using Price Feeds/l2-sequencer-feeds/l2-sequencer-flag.md

```diff
@@ -0,0 +1,157 @@
+
+The diagram below shows how these feeds update and how a consumer retrieves the status of the Arbitrum sequencer.
+
+![L2 Sequencer Feed Diagram](/images/data-feed/l2-diagram-arbitrum.png)
+
+1. Chainlink nodes trigger an OCR round every 30s and update the sequencer status by calling the `validate` function in the [`ArbitrumValidator` contract](https://github.com/smartcontractkit/chainlink/blob/master/contracts/src/v0.8/dev/ArbitrumValidator.sol) by calling it through the [`ValidatorProxy` contract](https://github.com/smartcontractkit/chainlink/blob/master/contracts/src/v0.8/ValidatorProxy.sol).
+1. The `ArbitrumValidator` checks to see if the latest update is different from the previous update. If it detects a difference, it places a message in the [Arbitrum inbox contract](https://developer.offchainlabs.com/docs/inside_arbitrum#the-big-picture).
+1. The inbox contract sends the message to the [`ArbitrumSequencerUptimeFeed` contract](https://github.com/smartcontractkit/chainlink/blob/master/contracts/src/v0.8/dev/ArbitrumSequencerUptimeFeed.sol). The message calls the `updateStatus` function in the `ArbitrumSequencerUptimeFeed` contract and updates the latest sequencer status to 0 if the sequencer is up and 1 if it is down. It also records the block timestamp to indicate when the message was sent from the L1 network.
+1. A consumer contract on the L2 network can read these values from the [`ArbitrumUptimeFeedProxy` contract](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.6/EACAggregatorProxy.sol), which reads values from the `ArbitrumSequencerUptimeFeed` contract.
+
+### Handling Arbitrum outages
+
+If the Arbitrum network becomes unavailable, the `ArbitrumValidator` contract continues to send messages to the L2 network through the delayed inbox on L1. This message stays there until the sequencer is back up again. When the sequencer comes back online after downtime, it processes all transactions from the delayed inbox before it accepts new transactions. The message that signals when the sequencer is down will be processed before any new messages with transactions that require the sequencer to be operational.
+
+### Example code for Arbitrum
+
+Create your consumer contract for uptime feeds similarly to contracts you would use for other Chainlink data feeds. You will need the following items:
+
+- The [sequencer uptime feed proxy address](#arbitrum): This contract on the L2 network
+- A contract that imports the [`AggregatorV2V3Interface.sol` contract](https://github.com/smartcontractkit/chainlink/blob/master/contracts/src/v0.8/interfaces/AggregatorV2V3Interface.sol): Use this interface to create a `sequencerUptimeFeed` object that points to the sequencer feed proxy.
+
+You can use the sequencer uptime feed on Arbitrum, but this example uses an Ethereum Rinkeby testnet feed for development and testing purposes.
+
+```solidity
+{% include 'samples/PriceFeeds/ArbitrumPriceConsumer.sol' %}
+```
+
```

Comment:
Replace 

```
This example includes a modified `getLatestPrice` function that reverts if the `STALENESS_THRESHOLD` is exceeded
```

with

```
This example includes a modified `getLatestPrice` function that reverts if the `GRACE_PERIOD_TIME` has not elapsed since the Sequencer has come back up.
```

Reply:
Done